pkgs/libresoc/soc.nix

```diff
@@ -0,0 +1,31 @@
+{
+  python39Packages,
+  fetchgit,
+  yosys,
+  libresoc-c4m-jtag,
+  libresoc-ieee754fpu,
+  libresoc-openpower-isa,
+  nmigen-soc,
+  power-instruction-analyzer,
+  pytest-output-to-files,
+}:
+with python39Packages;
+  buildPythonPackage {
+    pname = "soc";
+    version = "unstable-2024-03-31";
+
+    src = fetchgit {
+      url = "https://git.libre-soc.org/git/soc.git";
```

Comment:
upstream project that results in a python model of the target circuit (in nmigen/amaranth)